### PR TITLE
fix: (Platform) Value help dialog - styles mismatching

### PR DESCRIPTION
**Pull request by mykolavorobiov**
#### Please provide a link to the associated issue.

https://app.zenhub.com/workspaces/fundamental-library-5f32933d64202b0012e3572e/issues/sap/fundamental-ngx/4594

https://github.com/SAP/fundamental-ngx/issues/4594

#### Please provide a brief summary of this pull request.

Fixed styles mismatching and related behaviour with conditions

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples

### CODE REVIEW

File: libs/platform/src/lib/components/value-help-dialog/components/define-tab/define-tab.component.ts [hunk 1]
```diff
@@ -110,91 +111,53 @@ export class DefineTabComponent<T> extends VhdBaseTab implements OnInit, OnChang
     return undefined;
   }
 
+  _onSelectStrategy(): void {
```
Comment (kavya-b) [inline libs/platform/src/lib/components/value-help-dialog/components/define-tab/define-tab.component.ts:114]: `@hidden` and private may be needed?

File: libs/platform/src/lib/components/value-help-dialog/components/select-tab/select-tab.component.scss [hunk 2]
```diff
@@ -3,47 +3,55 @@ $block: fdp-value-help-dialog;
...
-            white-space: normal;
-            .#{$block}-mobile & {
-                justify-content: flex-start;
-            }
-        }
+        background-color: #f7f7f7;
+        background-color: var(--sapBackgroundColor, #f7f7f7);
     }
+
+    // &__advanced-search {
```
Comment (kavya-b) [inline libs/platform/src/lib/components/value-help-dialog/components/select-tab/select-tab.component.scss:10]: can we remove this commented code?

File: libs/platform/src/lib/components/value-help-dialog/value-help-dialog/value-help-dialog.component.scss [hunk 3]
```diff
@@ -1,37 +1,67 @@
...
         }
     }
     .fd-dialog__header {
         box-shadow: none;
+
         .fd-title {
             display: flex;
             align-items: center;
-            padding: .25rem 2rem;
+            padding-left: 2rem;
```
Comment (kavya-b) [inline libs/platform/src/lib/components/value-help-dialog/value-help-dialog/value-help-dialog.component.scss:54]: common values like 2rem here can be reused through a variable.

**Comment by kavya-b** [comment]
On selecting all items in the table, the tokenizer overflows beyond the dialog:
image

File: libs/platform/src/lib/components/value-help-dialog/components/select-tab/select-tab.component.scss (hunk 2 above)
Comment (mykolavorobiov) [reply libs/platform/src/lib/components/value-help-dialog/components/select-tab/select-tab.component.scss]: Removed

File: libs/platform/src/lib/components/value-help-dialog/components/define-tab/define-tab.component.ts (hunk 1 above)
Comment (mykolavorobiov) [reply libs/platform/src/lib/components/value-help-dialog/components/define-tab/define-tab.component.ts:115]: Added

File: libs/platform/src/lib/components/value-help-dialog/components/define-tab/define-tab.component.html [hunk 4]
```diff
@@ -1,12 +1,13 @@
 <h5 fd-title class="define-condition-title" i18n="@@platformI18nValueHelpDialog.Define.Condition.Title">Product</h5>
 <ng-container *ngFor="let item of _conditions; trackBy: _trackByKeyAndType; index as i">
-    <fd-layout-grid class="condition-row">
+    <fd-layout-grid class="condition-row" [noGap]="true">
         <div fdLayoutGridCol="5">
-            <div fdLayoutGridRow>
+            <!-- <div fdLayoutGridRow> -->
```
Comment (kavya-b) [inline libs/platform/src/lib/components/value-help-dialog/components/define-tab/define-tab.component.html:5]: can the commented code be removed?

**Review by kavya-b — APPROVED** [review]
minor comment, LGTM otherwise.

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Few things:

1. "hide all filters" button is misaligned with filter inputs, missing `0.25rem` padding bottom from the specs:
Screen Shot 2021-02-16 at 10 16 56 AM

2. Bottom right clear button should be compact, as well as the tokenizer
Screen Shot 2021-02-16 at 10 39 36 AM

3. In the specs the table does not have vertical column borders, you can use `[noBorderY]` input for that

4. The linked issue says that the specs do not mention pagination but it is still present here... is infinite scroll supposed to be used instead?

5. I would think that these two need to be left aligned, not centered
Screen Shot 2021-02-16 at 10 43 57 AM

**Review by Betrozov — APPROVED** [review]

**Review by JKMarkowski — APPROVED** [review]
LGTM

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by Seamoo13** [comment]
- [ ] 1. for define conditions tab and custom strategy labels examples, styles don't match. borders cutoff/hidden
Screenshot at 2021-04-01 17-36-35

Screenshot at 2021-04-01 17-35-36

Screenshot at 2021-04-01 17-43-40

**Review by Seamoo13 — APPROVED** [review]

**Pull request merged**